qt_ui/widgets/map/model/mapmodel.py

```diff
@@ -374,6 +380,12 @@ def reset_map_zones(self) -> None:
     def mapZones(self) -> NavMeshJs:
         return self._map_zones
 
+    def on_package_change(self) -> None:
+        if self.game_model.game is None:
+            return
+        self.game_model.game.compute_unculled_zones()
```

Comment:
There really isn't a good place to do this, is there. It's probably slightly better to do this in `AtoModel`, so if multiple things are listening for this event they don't have to coordinate (or duplicate) this. I don't like that the UI is driving this at all but I don't think we have a good way to do otherwise atm.

I had always thought that the fix here would be to move the culling zone calculation out of turn initialization entirely and instead do it during mission generation, but hadn't thought about the fact that the UI needs it. That definitely makes it harder.

Reply:
My decision for placing it in `MapModel` was due to the `unculledZonesChanged` signal. Most of the other options seemed like they would create a confusing/redundant signal chain.

Moved it to `AtoModel`. Can't imagine there would be harm in calling ASAP - what would ever require the updated packages along with the outdated unculled zones?